de.fu_berlin.inf.dpp.whiteboard/frontend/src/shapes/attributesmixin.js

```diff
@@ -0,0 +1,203 @@
+import SVG from "../constants/SVG";
+
+/**
+ * adds the attributes management needed to make a fabricjs object easily transferable.
+ * NOTE: all functions and attributes start with '$' to separate them from native fabricjs
+ * functions and properties
+ * @param {fabric.Object} superClass
+ * the class which will be extended and have these attributes
+ */
+let AttributeMixin = (superClass) => class extends superClass {
+  constructor(...params) {
+    super(...params);
+    //this objects attributes
+    this.$attrs = {};
+    //this object can be overwritten by inherited objects to define custom mapping of attributes
+    this.$objectSpecificAttributes = {};
+  }
+
+  /**
+   * returns the attribute with the given id
+   * @param {string} id
+   * id of the target attribute
+   * @returns {Object}
+   * attribute with the given id
+   */
+  $getAttr(id) {
+    return this.$attrs[id];
+  }
+
+  /**
+   * returns attribute's id of the attribute with the given name
+   * @param {string} name
+   * attribute's name
+   * @returns {string}
+   * attribute's id
+   */
+  $getAttrID(name) {
+    for (let attr of Object.values(this.$attrs))
+      if (attr.name == name)
+        return attr.rid;
+  }
+
+  /**
+   * returns array of attributes which can be directly given to java
+   * to be sent to other users
+   * @returns {{name: string, chdata: string, rid: string}[]}
+   * array of this object's attributes
+   */
+  $getTransferableAttrs() {
+    this.$refreshAttrs();
+    let toSend = [];
+    for (let attr of Object.values(this.$attrs))
+      if (attr.isDirty) {
+        toSend.push({ name: attr.name, chdata: attr.chdata.toString(), rid: attr.rid });
+        delete attr.isDirty;
```

Comment:
you already changed `delete attr.isDirty` => `attr.isDirty = false` in your previous commit. Will you change it here?